rust/template/differential_datalog/valmap.rs

```diff
@@ -90,8 +90,12 @@ impl<V: Display + Ord + Clone> DeltaMap<V> {
         for (relid, map) in &self.map {
             w.write_fmt(format_args!("{}:\n", R::relid2name(*relid).unwrap()))?;
             for (val, weight) in map {
-                w.write_fmt(format_args!("{}\n", *val))?;
-                assert_eq!(*weight, 1, "val={}, weight={}", *val, *weight);
+                if *weight == 1 {
+                    w.write_fmt(format_args!("{}\n", *val))?;
```

Comment:
Doesn't this case make life harder for everyone involved?

Reply:
This makes sure that output _sets_ are always be printed without multiplicities, which are always "+1".